app/js/UrlFetcher.js

```diff
@@ -76,15 +77,16 @@ module.exports = UrlFetcher = {
 
     return urlStream.on('response', function (res) {
       if (res.statusCode >= 200 && res.statusCode < 300) {
-        const fileStream = fs.createWriteStream(filePath)
+        const atomicWrite = filePath + '~'
```

Comment:
I think it would be possible for two users to download the same file at the same time, so either we live with the conflict or use a unique suffix here.

If/when we update this file to use fetch and return a promise then we could return the existing promise for any file that is already being downloaded.

Reply:
Good point. Too bad that there is no `fs.mktemp`. I'd rather not implement this myself, but go your proposed path of de-duplicating the download.

Reply:
Done in c8715ad644c29a1e4936fe395cc52b0c2a12f6e2

Reply:
Done in c8715ad644c29a1e4936fe395cc52b0c2a12f6e2